maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/DownloadSchemaRegistryMojo.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package io.confluent.kafka.schemaregistry.maven;
+
+import io.confluent.kafka.schemaregistry.client.SchemaMetadata;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaParseException;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Mojo(name = "download")
+public class DownloadSchemaRegistryMojo extends SchemaRegistryMojo {
+
+  @Parameter(required = false, defaultValue = ".avsc")
+  String schemaExtension;
+  @Parameter(required = false)
+  List<String> subjectPatterns;
+  @Parameter(required = true)
+  File outputDirectory;
+
+  @Parameter(required = false, defaultValue = "true")
+  boolean prettyPrintSchemas;
+
+  public DownloadSchemaRegistryMojo() {
+    this.subjectPatterns = new ArrayList<>();
+    this.subjectPatterns.add("^.+$");
+  }
+
+  Map<String, Schema> downloadSchemas(Collection<String> subjects) throws MojoExecutionException {
+    Map<String, Schema> results = new LinkedHashMap<>();
+
+    for (String subject : subjects) {
+      Schema.Parser parser = new Schema.Parser();
+      SchemaMetadata schemaMetadata;
+      try {
+        getLog().info(String.format("Downloading latest metadata for %s.", subject));
+        schemaMetadata = this.client().getLatestSchemaMetadata(subject);
+      } catch (RestClientException | IOException ex) {
+        throw new MojoExecutionException(
+            String.format("Exception thrown while downloading metadata for %s.", subject),
+            ex
+        );
+      }
+
+      try {
```

Comment:
Not really a big deal, but you can put this and the previous try/catch in one block -- `SchemaParseException` isn't a `RestClientException` or `IOException`, so there shouldn't be any conflicts.